src/HTTP.jl

```diff
@@ -37,24 +37,44 @@ include("sniff.jl")                    ;using .Sniff
 include("multipart.jl")                ;using .Forms
 include("Parsers.jl")                  ;import .Parsers: Headers, Header,
                                                          ParseError
-include("ConnectionPool.jl")           ;using .ConnectionPool
+include("Connections.jl")              ;using .Connections
+# backwards compat
+const ConnectionPool = Connections
 include("StatusCodes.jl")              ;using .StatusCodes
 include("Messages.jl")                 ;using .Messages
 include("cookies.jl")                  ;using .Cookies
 include("Streams.jl")                  ;using .Streams
+
+getrequest(r::Request) = r
+getrequest(s::Stream) = s.message.request
+
+# Wraps client-side "layer" to track the amount of time spent in it as a request is processed.
+function observe_layer(f)
+    function observelayer(req_or_stream; kw...)
+        req = getrequest(req_or_stream)
```

Comment:
could this be `nothing`?

Reply:
No, we're guaranteed to always have a Request here.